data-prepper-plugins/sqs-source/src/main/java/org/opensearch/dataprepper/plugins/source/sqssource/SqsSource.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.source.sqssource;
+
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.acknowledgements.AcknowledgementSetManager;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.plugin.PluginFactory;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.model.source.Source;
+import org.opensearch.dataprepper.plugins.aws.sqs.common.BufferAccumulator;
+import org.opensearch.dataprepper.plugins.aws.sqs.common.codec.Codec;
+import org.opensearch.dataprepper.plugins.aws.sqs.common.codec.JsonCodec;
+import org.opensearch.dataprepper.plugins.aws.sqs.common.metrics.SqsMetrics;
+
+@DataPrepperPlugin(name = "sqs", pluginType = Source.class,pluginConfigurationType = SqsSourceConfig.class)
+public class SqsSource implements Source<Record<Event>> {
+
+    private final SqsSourceConfig sqsSourceConfig;
+
+    private final AcknowledgementSetManager acknowledgementSetManager;
+
+    private final PluginMetrics pluginMetrics;
+
+    private SqsSourceService sqsSourceService;
+
+    private final boolean acknowledgementsEnabled;
+
+    private final Codec codec;
```

Comment:
I think the Codec concept is not ideal for this type of source. It should output the string and we let users parse it using a processor (e.g. grok or parse_json). Probably it is best to output an event that looks like:

```
message: "<exact string value from SQS message>"
```

Reply:
Thanks @dlvenable  for your review,  we are getting follwing issue when we add the output string to JacksonLog.builder().withData(message).build()   so shall push string like below json

    { "message " : "<<exact string value from SQS message>>" }

Exception:
java.lang.IllegalArgumentException: Unable to convert data into an event

Reply:
@udaych20 , Thanks for the response. So you will need to make a very small map with a single `message` key.

We already have a helper to do this for you:

```
Event event = JacksonEvent.fromMessage(sqsMessage.getMessage());
```

Provide that `event` from the library to the caller of the library.

Here is an example of using that `fromMessage`.

https://github.com/opensearch-project/data-prepper/blob/59c48e29a25e15b16e38f4c118477a041e458121/data-prepper-plugins/common/src/main/java/org/opensearch/dataprepper/plugins/source/StdInSource.java#L81-L84

Reply:
Thanks @dlvenable for your response, Sure i will change the code accordingly.